### PR TITLE
Bugfix/duplicate sla event ids

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**
- https://github.com/fivetran/dbt_zendesk/issues/108
- https://github.com/fivetran/dbt_zendesk/issues/109

**This PR will result in the following new package version:**

v0.12.0

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**
- Updates the way business hour reply time records are getting brought into the final `sla__policies` model. Originally the logic in the upstream `int_zendesk__reply_time_combined` was not capturing all cases and therefore duplicate `sla_event_id`s were being persisted. 
- Additionally, since this an update off of the v0.11.0 release where the errors appeared, rather off the current release that rolled back to v0.10.2, this will have the changes from the v0.11.0 release plus the updates that fix for the errors. Namely the addition of the lagging_time_block and filtered_reply_times CTE in `int_zendesk__reply_time_combined` because of the inclusion of schedule holiday time in business hours calculations
- Also introduced these 2 new fields in  `int_zendesk__reply_time_business_hours` for later use downstream
--    ticket_created_at
--    sla_schedule_end_at

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (if applicable)
using_schedules = True and using_schedules = False

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing 

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

- Joe replicated the error in our dev Zendesk account and were able to get the right records to be brought through
- Worked through sample data that customer sent us and confirmed that only the right records were brought through
- Customer also tested this branch and confirmed these changes brought the right records through and resolved the fanouts

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR 
    
- [x] README updates have been applied (if applicable)
    
- [ ] DECISIONLOG updates have been updated (if applicable)
- [ ] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [ ] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
☠️

### CODE REVIEW

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli these changes look great! I made a few updates myself to pick up changes that were missing from the original 0.11.0 and were not part of the problematic code. Additionally, I made some minor edits in the sla policy model which I came across when doing the review and noticed some discrepancies when validating. Would you be able to review the changes I made and let me know if you have any concerns or comments.

Your changes look great though! I have just a few very minor comments I would like you to respond before approving.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,12 +2,16 @@
 
 This release reverts the rollback of v0.11.2 and fixes the issues introduced in v0.11.0-v0.11.1 by the addition of `schedule_holidqay`.
```
[the remark this answers is not in the document]
Comment (fivetran-reneeli) [reply CHANGELOG.md]: That's true, it's mentioned in the below bullet points

[the remark this answers is not in the document]
Comment (fivetran-reneeli) [reply CHANGELOG.md]: Oh wait, it looks like this was already updated

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Thanks for working through these updates @fivetran-reneeli! These changes look good after my validation and review. This is good to be merged into the base branch.

I will then conduct the final review there.

**Pull request merged**